src/compiler/checker.ts

```diff
@@ -9184,6 +9198,16 @@ namespace ts {
             return links.type;
         }
 
+        function getWriteTypeOfSymbol(symbol: Symbol): Type {

```

Comment:
This function and the associated SymbolLinks cache promise a *tiny* too much about write types. It doesn't quite look like the ad-hoc rule that it is. Fine for now, has a small chance of training a naive reader in a few years time into adding ad-hoc write types elsewhere.

(This extends to the way that `getTypeOfSetAccessor` delegates to `resolveTypeOfAccessors` instead of the other way round, although get/set checking is extremely interleaved in the latter, so it'd be hard to change.)

Reply:
Ideas - maybe just rename it?

Reply:
maaaybe `getSetAccessorTypeOfSymbol`. That at least sounds clunky and specific.

Reply:
Renamed